lib/libvcc/vcc_compile.c

```diff
@@ -580,26 +586,53 @@ vcc_resolve_includes(struct vcc *tl)
 				vcc_ErrWhere(tl, t1);
 				return;
 			}
-			vsb = VSB_new_auto();
-			AN(vsb);
 			p = strrchr(t1->src->name, '/');
 			AN(p);
 			VSB_bcat(vsb, t1->src->name, p - t1->src->name);
 			VSB_cat(vsb, t1->dec + 1);
-			AZ(VSB_finish(vsb));
-			sp = vcc_file_source(tl, VSB_data(vsb));
-			VSB_destroy(&vsb);
+		} else
+			VSB_cat(vsb, t1->dec);
+
+		AZ(VSB_finish(vsb));
+
+		/* only try to glob if the path starts with "./" or "/", and if 
+		 * it contains a "*"
+		 */
+		if (((t1->dec[0] == '.' && t1->dec[1] == '/') ||
+		    t1->dec[0] == '/') && strchr(VSB_data(vsb), '*')) {
+			if (glob(VSB_data(vsb), 0, NULL, &glb)) {
+				VSB_printf(tl->sb,
+				    "Globbing pattern matched no file (%s)\n",
+				    VSB_data(vsb));
```

Comment:
assert that the `glob()` return is `GLOB_NOMATCH` ?

Reply:
and wouldn't the right way be to attempt including the pathname with the literal wildcard?

Reply:
> and wouldn't the right way be to attempt including the pathname with the literal wildcard?

sorry, I didn't catch your meaning here

Reply:
like shell attempts to use the file with the literal asterisk when the glob expansion fails

```slink@haggis:~$ ls *guillaume*
ls: cannot access '*guillaume*': No such file or directory
slink@haggis:~$ touch *guillaume*
slink@haggis:~$ ls *guillaume*
*guillaume*
```

(and no, I do not mean to `touch` you no matter how many stars around you ;) )

Reply:
ah, good point!

Reply:
actually, no that's unnecessary since the string `*guillaume*` is a match for the globbing pattern `*guillaume*`. If globbing failed, there's nothing more we can do. Or did I miss something?

Reply:
I see your point @gquintard , my example was bad.
I guess `[ab].vcl` is a better one, this should, upon failure to expand, attempt to open a file by that literal name.

Reply:
right, `*` isn't the only possible pattern. I'll adjust